rust/lance-encoding/src/configured_compression_strategy.rs

```diff
@@ -0,0 +1,470 @@
+        if params.compression.as_deref() == Some("none") {
+            return Ok(Box::new(BinaryMiniBlockEncoder::default()));
+        }
+
+        // 2. Choose base encoder (FSST or Binary)
+        let mut base_encoder: Box<dyn MiniBlockCompressor> = if max_len
+            >= FSST_LEAST_INPUT_MAX_LENGTH
+            && data_size >= FSST_LEAST_INPUT_SIZE as u64
+        {
+            Box::new(FsstMiniBlockEncoder::default())
+        } else {
+            Box::new(BinaryMiniBlockEncoder::default())
+        };
+
+        // 3. Apply general compression if configured
+        if let Some(compression_scheme) = &params.compression {
+            if compression_scheme != "none" {
+                let scheme: CompressionScheme = compression_scheme.parse()?;
+                let config = CompressionConfig::new(scheme, params.compression_level);
+                base_encoder = Box::new(GeneralMiniBlockCompressor::new(base_encoder, config));
+            }
+        }
+
+        Ok(base_encoder)
+    }
+}
+
+impl CompressionStrategy for ConfiguredCompressionStrategy {
+    fn create_miniblock_compressor(
+        &self,
+        field: &Field,
+        data: &DataBlock,
+    ) -> Result<Box<dyn MiniBlockCompressor>> {
+        // Get merged parameters for this field
+        let params = self
+            .params
+            .get_field_params(&field.name, &field.data_type());
+
+        // Route to appropriate builder based on data type
+        match data {
+            DataBlock::FixedWidth(fixed_width) => {
+                self.build_fixed_width_compressor(&params, field, fixed_width)
+            }
+            DataBlock::VariableWidth(variable_width) => {
+                self.build_variable_width_compressor(&params, variable_width)
+            }
+            DataBlock::Struct(_) => {
+                // Struct compression doesn't use parameters currently
+                Ok(Box::new(PackedStructFixedWidthMiniBlockEncoder::default()))
+            }
+            DataBlock::FixedSizeList(_) => {
+                // FSL doesn't support compression currently
+                Ok(Box::new(ValueEncoder::default()))
+            }
+            _ => {
+                // Fall back to default strategy for unsupported types
+                self.default_strategy
+                    .create_miniblock_compressor(field, data)
+            }
+        }
+    }
+
+    fn create_per_value(
+        &self,
+        field: &Field,
+        data: &DataBlock,
+    ) -> Result<Box<dyn PerValueCompressor>> {
+        // Per-value compression doesn't use parameters currently
+        self.default_strategy.create_per_value(field, data)
+    }
+
+    fn create_block_compressor(
+        &self,
+        field: &Field,
+        data: &DataBlock,
+    ) -> Result<(Box<dyn BlockCompressor>, pb::ArrayEncoding)> {
+        // Block compression doesn't use parameters currently
+        self.default_strategy.create_block_compressor(field, data)
+    }
+}
```

Comment:
This code seems pretty similar to what's already in `DefaultCompressionStrategy`.  Instead of making a new `ConfiguredCompressionStrategy` can we just modify the existing default strategy to use compression params?

Reply:
I added `ConfiguredCompressionStrategy` to avoid the existing logic, but I'm perfectly fine with just adding it to `DefaultCompressionStrategy` instead. I'll go ahead and make this change.